commit f7a65de37ddc759283308fcad97681aebcb1d2da
Author: Miroslav Rezanina <mrezanin@redhat.com>
Date:   Wed Apr 3 08:44:26 2013 +0200

    qxl: better vga init in enter_vga_mode
    
    Ask the vga core to update the display.  Will trigger dpy_gfx_resize
    if needed.  More complete than just calling dpy_gfx_resize.
    
    Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
    (cherry picked from commit c099e7aa0295678859d58e9e60b7619f6ae3bac8)

diff --git a/hw/qxl.c b/hw/qxl.c
index 436e375..80bd92a 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -1073,8 +1073,8 @@ static void qxl_enter_vga_mode(PCIQXLDevice *d)
     trace_qxl_enter_vga_mode(d->id);
     qemu_spice_create_host_primary(&d->ssd);
     d->mode = QXL_MODE_VGA;
-    dpy_gfx_resize(d->ssd.ds);
     vga_dirty_log_start(&d->vga);
+    vga_hw_update();
 }
 
 static void qxl_exit_vga_mode(PCIQXLDevice *d)
